disnake/guild.py

```diff
@@ -2163,6 +2163,13 @@ async def create_scheduled_event(
 
         Creates a :class:`GuildScheduledEvent`.
 
+        If ``channel`` is provided and has a type matching to a type of :class:`GuildScheduledEventEntityType`,
```

Comment:
This might be better represented using a table:
(once again can't directly add a suggestion)
```diff
diff --git a/disnake/guild.py b/disnake/guild.py
index 795e7553..60579ca3 100644
--- a/disnake/guild.py
+++ b/disnake/guild.py
@@ -2163,18 +2163,17 @@ class Guild(Hashable):
 
         Creates a :class:`GuildScheduledEvent`.
 
-        If ``channel`` is provided and has a type matching to a type of :class:`GuildScheduledEventEntityType`,
-            the ``entity_type`` is not required and will automatically match.
+        Based on the channel/entity type, there are different restrictions regarding
+        other parameter values, as shown in this table:
 
-        If ``channel`` is not provided, then ``entity_type`` must be provided.
-
-        If ``channel`` is None, then ``entity_type`` is assumed to be :class:`GuildScheduledEventEntityType.external`
+        .. csv-table::
+            :header: "``channel``", "``entity_type``", "``scheduled_end_time``", "``entity_metadata`` with location"
 
-        If ``entity_type`` is :class:`GuildScheduledEventEntityType.external`:
+            :class:`VoiceChannel`, automatically set to :attr:`~GuildScheduledEventEntityType.voice`,          optional, unset
+            :class:`StageChannel`, automatically set to :attr:`~GuildScheduledEventEntityType.stage_instance`, optional, unset
+            unspecified snowflake, required,                                                                   optional, unset
+            ``None`` or unset,     automatically set to :attr:`~GuildScheduledEventEntityType.external`,       required, required
 
-        - ``channel`` should not be set
-        - ``entity_metadata`` with a location field must be provided
-        - ``scheduled_end_time`` must be provided
 
         .. versionadded:: 2.3
 

```

I'm a bit unsure about the "unspecified snowflake" phrasing though, perhaps someone else has a better idea for that.
In any case, looks like this:
![image](https://user-images.githubusercontent.com/8530778/180074199-b0a909bb-a5cb-4292-9859-341a69422dfa.png)

Reply:
changed in [fce1fd9](https://github.com/DisnakeDev/disnake/pull/617/commits/fce1fd99f4eb82473fad14e3ce3a417d5fd88e4e)